packages/autocomplete-core/src/utils/__tests__/getNormalizedSources.test.ts

```diff
@@ -89,9 +89,31 @@ describe('getNormalizedSources', () => {
 
     // @ts-expect-error
     await expect(getNormalizedSources(getSources, params)).rejects.toEqual(
-      new Error(
-        '[Autocomplete] The `getSources` function must return a `sourceId` string but returned type "undefined":\n\nundefined'
-      )
+      new Error('[Autocomplete] A source must provide a `sourceId` string.')
+    );
+  });
+
+  test('with wrong `sourceId` type triggers invariant', async () => {
```

Comment:
Since we use TypeScript I don't think we need to unit test wrong types. The test code base is gonna grow a lot otherwise.

Reply:
Aren't we also testing the return type of `getSources`?

Reply:
Right, that's an exception. We do it because that's a sensible code path where users might mess up easily (and also plugin authors) due to the complex type. A string is a simple type.

Reply:
I now see that you've changed the invariant condition so this test makes sense 👍 (it used to check for `undefined`).